### ISSUE TITLE
to_number does not correctly handle floating point numbers without dots

### ISSUE THREAD

**Issue by jtprobst** [jmespath/jmespath.py#120]
The [JMESPath specification of the to_number function](http://jmespath.readthedocs.io/en/latest/specification.html#to-number) states that everything matching the `json-number` production must be supported. However, this implementation will handle floating point numbers incorrectly if they do not contain a dot.

Please consider the following example
```json
{"x": "1e+21"}
```

When searched with the JMESPath
```
x | to_number(@)
```

this implementation will return None since the number is not recognized as such. The parser will only try parsing a floating point number if it contains a dot, as can be seen [in the code](https://github.com/jmespath/jmespath.py/blob/develop/jmespath/functions.py#L205)

Therefore, the following (incorrect) behavior can be observed
```python
$ python
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import jmespath
>>> data = {"x": "1e+21"}
>>> jmespath.search("x | to_number(@)", data)
>>> 
```

Please note the None return value which is not printed by the REPL.

The JSON standard allows such numbers, as per http://www.json.org/ where the grammar allows a special case of number `number := int exp` where `exp := e digits` and `e := 'e+'`.

My package version is 0.9.1.

**Comment by jamesls** [comment]
Thanks for reporting.  This is fixed in #134.

**Comment by jtprobst** [comment]
@jamesls Thank you very much for your efforts! It is much appreciated!
